chroma-manager/tests/framework/integration/installation_and_upgrade/run_tests

```diff
@@ -225,18 +275,74 @@ LOG_LEVEL = logging.DEBUG
 $LOCAL_SETTINGS
 EOF1
 
+# override /usr/share/chroma-manager/storage_server.repo
+if [ -f /tmp/storage_server.repo ]; then
+    cp /tmp/storage_server.repo /usr/share/chroma-manager/storage_server.repo
+fi
+
 # Ensure that coverage is disabled
 # https://github.com/pypa/virtualenv/issues/355
 python_version=\$(python -c 'import platform; print \".\".join(platform.python_version_tuple()[0:2])')
 rm -f /usr/lib/python\$python_version/site-packages/sitecustomize.py*"
 
+# the manual bits that need to be done for 3.x to 4.x upgrades
+# see https://github.com/intel-hpdd/intel-manager-for-lustre/issues/125
+pdsh -l root -R ssh -S -w $(spacelist_to_commalist ${STORAGE_APPLIANCES[@]}) "exec 2>&1; set -xe
+yum -y install epel-release
+yum -y install pdsh
+mv /etc/yum.repos.d/Intel-Lustre-Agent.repo{,.iml-3.x}" | dshbak -c
+if [ ${PIPESTATUS[0]} != 0 ]; then
+    exit 1
+fi
+
+ssh root@"$CHROMA_MANAGER" "exec 2>&1; set -xe
+yum -y install pdsh
+pdcp -l root -R ssh -w $(spacelist_to_commalist "${STORAGE_APPLIANCES[@]}") /usr/share/chroma-manager/storage_server.repo /etc/yum.repos.d/Intel-Lustre-Agent.repo"
+
+pdsh -l root -R ssh -S -w "$(spacelist_to_commalist "${STORAGE_APPLIANCES[@]}")" "exec 2>&1; set -xe
+sed -ne '/\[iml-agent\]/,/^$/p' /etc/yum.repos.d/Intel-Lustre-Agent.repo.iml-3.x >> /etc/yum.repos.d/Intel-Lustre-Agent.repo
+# TODO: we really just need to adjust cache timeouts here to account for
+#       our compressed timelines
+yum clean all
+yum-config-manager --enable iml-agent
+
+yum -y upgrade chroma-agent\*
+
+systemctl stop zed
```

Comment:
Is zed actually running in 3.1?

Reply:
I dunno.  I'm just implementing what is documented in #125.